lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -161,6 +168,40 @@ defmodule Bamboo.SMTPAdapter do
     |> add_smtp_line(text_body)
   end
 
+  defp add_multipart_mixed_split(body, delimiter) do
+    body
+    |> add_smtp_line(delimiter)
+  end
+
+  defp add_attachment_header(body, attachment) do
+    << random :: size(32) >> = :crypto.strong_rand_bytes(4)
+    body
+    |> add_smtp_line("Content-Type: #{attachment.content_type}; name=\"#{attachment.filename}")
+    |> add_smtp_line("Content-Disposition: attachment; filename=\"#{attachment.filename}")
+    |> add_smtp_line("Content-Transfer-Encoding: base64")
+    |> add_smtp_line("X-Attachment-Id: #{random}")
+  end
+
+  defp add_attachment_body(body, data) do
+    add_smtp_line(body, Base.encode64(data))
+  end
+
+  defp add_attachment(nil, _), do: ""
+  defp add_attachment(attachment, multi_part_mixed_delimiter) do
+    ""
+    |> add_multipart_mixed_split(multi_part_mixed_delimiter)
+    |> add_attachment_header(attachment)
+    |> add_smtp_line("")
+    |> add_attachment_body(attachment.data)
```

Comment:
It doesn't exist _yet_ :) This was done as a preparation for this pull request on Bamboo: https://github.com/thoughtbot/bamboo/pull/292

More discussion here: https://github.com/thoughtbot/bamboo/issues/286